File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let x = 0;
    let y = matrix[0].length - 1;
    
    while(x < matrix.length && y >= 0) {
        if(matrix[x][y] == target) return true;
        
        if(matrix[x][y] > target) {
            y--;
        } else {
            x++;
        }
    }
    
    return false;
};

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        int n = matrix[0].size();
        int r = 0, c = n-1;
        
        while(r < m && c >= 0) {
            int cur = matrix[r][c];
            if(cur == target) return true;
            if(cur < target) {
                r++;
            } else {
                c--;
            }
        }
        
        return false;
    }
};